docs/protocol.md

```diff
@@ -1062,6 +1063,118 @@ RPC
 ```
 </details>
 
+### Catching up on past messages on a channel using paging
+
+🧭 **RPC Message** > **Query** > **Paged Catchup**
+
+By executing a paged catchup action, a client can ask the server to receive a specified number of
+past messages on a specific channel.
+
+For now, it is to be used to retrieve chirps on the social media channel by paging when a new client joins the LAO instead of getting all chirps at once. This is done in an effort to reduce network traffic at catchup.
+
+This may serve as a starting point for the paging of messages in other channels as a future optimization.
+
+RPC 
+
+```json5
+// ../protocol/examples/query/paged_catchup/paged_catchup.json
+
+{
+    "jsonrpc": "2.0",
+    "id": 3,
+    "method": "paged_catchup",
```

Comment:
Should it be a general catchup message? If so, how do you order non-chirp messages? Do you plan to add a timestamp to all messages?

Reply:
The focus for this semester is on chirps, but if we can have one message that can then be used by future generations to extend paging to other types of messages, then that would be great.